gettsim/transfers/grunds_im_alter.py

```diff
@@ -0,0 +1,425 @@
+    grunds_im_alter_m_minus_eink_hh: FloatSeries,
+    alle_erwachsene_rentner_hh: BoolSeries,
+    vermögen_hh: FloatSeries,
+    vermög_freib_grunds_im_alter_hh: FloatSeries,
+) -> FloatSeries:
+    """Calculate Grundsicherung im Alter on household level.
+
+    # ToDo: There is no check for Wohngeld included as Wohngeld is
+    # ToDo: currently not implemented for retirees.
+
+    # ToDo: Grundsicherung im Alter is only payed if all adults in the household
+    # ToDo: are retired (This is a simplification by GETTSIM)
+
+    Parameters
+    ----------
+    grunds_im_alter_m_minus_eink_hh
+        See :func:`grunds_im_alter_m_minus_eink_hh`.
+    alle_erwachsene_rentner_hh
+        See :func:`alle_erwachsene_rentner_hh`.
+
+    Returns
+    -------
+
+    """
+    out = grunds_im_alter_m_minus_eink_hh.copy()
+
+    # Wealth check
+    out.loc[vermögen_hh >= vermög_freib_grunds_im_alter_hh] = 0
+
+    # Only pay Grundsicherung im Alter if all adults are retired (simplification
+    # by GETTSIM)
+    out.loc[~alle_erwachsene_rentner_hh] = 0
+    return out
+
+
+def grunds_im_alter_m_minus_eink_hh(
+    regelbedarf_m_hh: FloatSeries,
+    mehrbedarf_behinderung_m_hh: FloatSeries,
+    kindergeld_m_hh: FloatSeries,
+    unterhaltsvors_m_hh: FloatSeries,
+    anr_eink_grunds_im_alter_m_hh: FloatSeries,
+) -> FloatSeries:
+    """Calculate remaining basic subsistence after subtracting income
+    and other benefits.
+
+    Parameters
+    ----------
+    regelbedarf_grunds_im_alter_vermögens_check_m_hh
+        See :func:`regelbedarf_grunds_im_alter_vermögens_check_m_hh`.
+    mehrbedarf_behinderung_m_hh
+        See :func:`mehrbedarf_behinderung_m_hh`.
+    kindergeld_m_hh
+        See :func:`kindergeld_m_hh`.
+    unterhaltsvors_m_hh
+        See :func:`unterhaltsvors_m_hh`.
+    anr_eink_grunds_im_alter_m_hh
+        See :func:`anr_eink_grunds_im_alter_m_hh`.
+
+    Returns
+    -------
+
+    """
+    out = (
+        regelbedarf_m_hh
+        + mehrbedarf_behinderung_m_hh
+        - anr_eink_grunds_im_alter_m_hh
+        - unterhaltsvors_m_hh
+        - kindergeld_m_hh
+    )
+    return out.clip(lower=0)
+
+
+def anr_eink_grunds_im_alter_m_hh(
+    anr_eink_grunds_im_alter_m: FloatSeries, hh_id: IntSeries
+) -> FloatSeries:
+    """Aggregate income which is considered in the calculation of Grundsicherung im
+    Alter on household level.
+
+    Parameters
+    ----------
+    anr_eink_grunds_im_alter_m
+        See :func:`anr_eink_grunds_im_alter_m`.
+    hh_id
+        See basic input variable :ref:`hh_id <hh_id>`.
+
+    Returns
+    -------
+
+    """
+    return anr_eink_grunds_im_alter_m.groupby(hh_id).sum()
+
+
+def anr_eink_grunds_im_alter_m(
+    anr_erwer_eink_grunds_im_alter_m: FloatSeries,
+    anr_priv_rente_grunds_im_alter_m: FloatSeries,
+    anr_ges_rente_grunds_im_alter_m: FloatSeries,
+    sonstig_eink_m: FloatSeries,
+    vermiet_eink_m: FloatSeries,
+    anr_kap_eink_grunds_im_alter_m: FloatSeries,
```

Comment:
@hmgaudecker 
> Alternatively, one could do erw_ and kap_, but we are using erw_ as an abbreviation for "Erwachsene" a lot, so I'd sort of like to leave it as is.

I changed it to `kap_eink` and `erwer_eink` now.

I also checked that the allowance for capital income does not apply to rental income. So it is correct to keep it separate here.

Reply:
Everywhere in the code? 

I really do not want to be searching for "erw_" in the codebase and decide on a case-by-case basis whether it is what I want to see or not...

Reply:
Could we discuss changes like that a little longer? I am in favor of reverting.

Reply:
Yes, everywhere. 

I picked `erwer_eink ` to avoid confusion with `erw_`

> Could we discuss changes like that a little longer?

Ok, will do next time. We can easily revert it now. I, however, like the shorter version as it avoids two purely private functions.

Reply:
I think the issue is with what we mean by "internal". In my view, it is definitely not "purely internal". Anybody (except for Stata users) could easily access them if they want. I agree that variable name length is not a great reason to exclude something from the list of directly documented functions. But the drawback are function names that are hard to parse. It would not occur to me to abbreviate `erwerbs` with `erwer`, other than this hard character limit. It will be used at many other points as well, is this local change really worth it? As for `kap_`, I would not want to rename it until we have answered:

*Btw, are we talking about the same definition of capital income everywhere? E.g., income from renting out property would not be kapitaleink_ for tax purposes, but maybe here?*

To summarize:

- There are clear trade-offs between the breadth of variables accessible to Stata users on the one hand and reasonable naming for all others and developers on the other hand. 
- How we solve them, we just have to do on a case-by-case basis
- In doing so, we should be sure to do global optimisation and not look locally at some portion of the code only -- hence the comment on different notions of capital income.

Thanks!!!

Reply:
> It will be used at many other points as well

For me this is also a reason to keep it short since it might lead to more private functions at other places.

But no need to discuss these variable names too much. Let's revert to `erwerbseink`!

> Btw, are we talking about the same definition of capital income everywhere? E.g., income from renting out property would not be kapitaleink_ for tax purposes, but maybe here?

As mentioned above (probably language was not precise enough), I checked that it does not cover rental income for Grundrente and Grundsicherung im Alter. I now also checked the other places where it is used. Everything seems to be consistent.

> I also checked that the allowance for capital income does not apply to rental income. So it is correct to keep it separate here.

Do you agree with ``kap_eink`` then?

Reply:
And thanks for the good summary! I am totally on the same page, but putting more weight on avoiding internal functions if possible (I think it is not only relevant for Stata users, but also a signal for Python users). I think having consistent names across modules is very important! But having abbreviations that one wouldn't come up usually is not a big problem as long as it works the other way around (when reading the abbreviation it is clear what is meant)

Reply:
> > I also checked that the allowance for capital income does not apply to rental income. So it is correct to keep it separate here.

Thanks, I did miss it above indeed! My bad, apologies.

> Do you agree with `kap_eink` then?

Do you mind leaving it as is / changing to `kapitaleink` for the moment? I am happy to revisit at some other point, but I do like the rough symmetry in lengths of the different income types.

For this particular case, one might even make a case for keeping both private even if short enough -- might add room for confusion if one is there but not the other.

Reply:
> I think it is not only relevant for Stata users, but also a signal for Python users

Fully agreed, of course. But if we make clear that the most important reason for making long variable names private is our love and care for Stata users, that signal might not be quite as strong as in other Python code...

> I think having consistent names across modules is very important! But having abbreviations that one wouldn't come up usually is not a big problem as long as it works the other way around (when reading the abbreviation it is clear what is meant)

For people who do not look at the codebase all the time, I'd argue that is different. This creates frictions when searching for things. 

But that is exactly why it needs good discussion.

Reply:
> Do you mind leaving it as is / changing to kapitaleink for the moment?

will do

We definitely need an agreement and better guidelines here. It is not feasible to discuss about each single column name in detail.

Btw. I also made the change `vermögensfreib` -> `vermög_freib` (globally). Would you like to revert this, too, for the moment?

Reply:
> > Do you mind leaving it as is / changing to kapitaleink for the moment?
> 
> will do

Thanks!

> We definitely need an agreement and better guidelines here. It is not feasible to discuss about each single column name in detail.

Indeed. This said, changes that affect the entire code base and not just a single variable just need discussion and some form of consensus, no way around it. However, down the road it is pretty clear this a) won't happen that much in the first place and b) we can't change variable names everywhere because some concatenated variable name would end up being too long. They will only become more numerous.

> Btw. I also made the change `vermögensfreib` -> `vermög_freib` (globally). Would you like to revert this, too, for the moment?

No, perfectly fine. We agreed on that over on #314 and just did not do it there because we noticed the inconsistency too late. All good.